### DISCUSSION TITLE
OK to power the RPi from a separate house circuit (not from the battery or controller)

### DISCUSSION THREAD

**Discussion by eBoon123** [jgyates/genmon#1207] [Questions]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).
- [X] I have included the generator make and model in my submission.

### Version

Latest

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have not submitted logs

### What is/are your question(s)?

I'm helping a friend get the software running but we have been having problems with the converter hanging. Currently powering the RPi from the battery and everything is rock solid - until the generator kicks on then about 25% of the time the adapter hangs. To get everything running again, we can simply disconnect power from the adapter and genmon starts talking to the controller again. Powering the adapter from the 3.3 output of the RPi made it better, but still hangs often. We are pulling the ground from the controller to the adapter.

Considering powering the RPi and adapter from a separate dedicated house circuit. Would there be any issues doing it that way?

Thanks in advance for any guidance you can offer.

**Comment by skipfire** [comment]
First question is what adapter are you using and how are you powering the Pi now?  Power from a house can be ok (a dedicated circuit certainly isn't necessary for the Pi, but depending on how you want to install might be for code), but the first thing is you will want a UPS or other backup power for it, otherwise when the outage starts the Pi will lose power and reboot, not necessarily a huge issue, but one to be aware of.  The next thing is I'd recommend a galvanically isolated interface (commonly opto-isolated) since they are on different power systems.  When the Pi is on the same power system as the generator it's fairly pointless as the power makes it non-isolated.

**Comment by jgyates** [comment]
also include what pi model you are using. The pi zero has some issues on a Nexus model when powering from the molex connector.

I try to make sure the converter and the pi have a common ground. If the pi power supply ground and the converter ground is not the same it can work but powering the generator can push a marginal (improperly grounded) system to failure. I have experienced that first hand.
